Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc

```diff
@@ -9,98 +9,104 @@ ________________________________________________________________**/
+}
 
-  thePCCob->eventCounter(bx);
+std::shared_ptr<reco::PixelClusterCounts> AlcaPCCIntegrator::globalBeginLuminosityBlockSummary(
+    edm::LuminosityBlock const&, edm::EventSetup const&) const {
+  return std::make_shared<reco::PixelClusterCounts>();
+}
 
-  //Looping over the clusters and adding the counts up
-  edm::Handle<reco::PixelClusterCountsInEvent> pccHandle;
-  iEvent.getByToken(pccToken_, pccHandle);
+void AlcaPCCIntegrator::globalEndLuminosityBlockSummary(edm::LuminosityBlock const&,
+                                                        edm::EventSetup const&,
+                                                        reco::PixelClusterCounts*) const {}
 
-  const reco::PixelClusterCountsInEvent inputPcc = *(pccHandle.product());
-  thePCCob->add(inputPcc);
+void AlcaPCCIntegrator::accumulate(edm::StreamID iID, const edm::Event& iEvent, const edm::EventSetup&) const {
+  const reco::PixelClusterCountsInEvent& inputPcc = iEvent.get(pccToken_);
+  //add the PCCs from the event to the stream cache
+  streamCache(iID)->push_back(inputPcc);
 }
 
-//--------------------------------------------------------------------------------------------------
-void AlcaPCCIntegrator::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) {
-  //PCC object at the beginning of each lumi section
-  thePCCob = std::make_unique<reco::PixelClusterCounts>();
-  countLumi_++;
+void AlcaPCCIntegrator::streamEndLuminosityBlockSummary(edm::StreamID iID,
+                                                        edm::LuminosityBlock const&,
+                                                        edm::EventSetup const&,
+                                                        reco::PixelClusterCounts* iCounts) const {
+  auto Counts = streamCache(iID);
+  for (auto& counts : *Counts) {
+    //add each stream's cache to the LuminosityBlock cache
+    iCounts->add(counts);
+  }
+  //now clear in order to be ready for the next LuminosityBlock
+  Counts->clear();
 }
 
-//--------------------------------------------------------------------------------------------------
-void AlcaPCCIntegrator::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup) {}
-
-//--------------------------------------------------------------------------------------------------
-void AlcaPCCIntegrator::endLuminosityBlockProduce(edm::LuminosityBlock& lumiSeg, const edm::EventSetup& iSetup) {
-  //Saving the PCC object
-  lumiSeg.put(std::move(thePCCob), std::string(prodInst_));
+void AlcaPCCIntegrator::globalEndLuminosityBlockProduce(edm::LuminosityBlock& iLumi,
+                                                        edm::EventSetup const&,
+                                                        reco::PixelClusterCounts const* iCounts) const {
+  std::unique_ptr<reco::PixelClusterCounts> lumiPCC = std::make_unique<reco::PixelClusterCounts>(*iCounts);
+  //save the PCC object
+  iLumi.put(lumiPutToken_, std::move(lumiPCC));
```

Comment:
This could be simplified to
```suggestion
  //save the PCC object
  iLumi.emplace(lumiPutToken_, *iCounts);
```